src/buffer.jl

```diff
@@ -0,0 +1,432 @@
+# Support for python buffer protocol buffers
+
+# for versions > 3.2 (TODO: obj field not in struct for python versions <= 3.2)
```

Comment:
@stevengj I had a look, that comment is just wrong.  Perhaps I wrote that initially based on the documentation ->
[Python 2.7](https://docs.python.org/2/c-api/buffer.html#bufferobjects) vs [Python 3.4](https://docs.python.org/3.4/c-api/buffer.html#bufferobjects) where the obj field is documentated in the latter.

It was a bit speculative at the time, but I really should finish this PR.  Bidirectional zero copy buffers would be a really nice feature.